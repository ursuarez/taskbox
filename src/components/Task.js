import PropTypes from 'prop-types';
import React from 'react';
import styled from 'styled-components';

import { colors } from './styles/colors';

const { taskMainColor } = colors;

const TaskContainer = styled.div`
	border: 1px solid ${taskMainColor};
`;

export default function Task({
	task: { id, title, state },
	onArchiveTask,
	onPinTask,
}) {
	return (
		<TaskContainer className={`list-item ${state}`}>
			<label className='checkbox'>
				<input
					type='checkbox'
					defaultChecked={state === 'TASK_ARCHIVED'}
					disabled={true}
					name='checked'
				/>
				<span className='checkbox-custom' onClick={() => onArchiveTask(id)} />
			</label>
			<div className='title'>
				<input
					type='text'
					value={title}
					readOnly={true}
					placeholder='Input title'
				/>
			</div>

			<div className='actions' onClick={(event) => event.stopPropagation()}>
				{state !== 'TASK_ARCHIVED' && (
					<a onClick={() => onPinTask(id)}>
						<span className={`icon-star`} />
					</a>
				)}
			</div>
		</TaskContainer>
	);
}

Task.propTypes = {
	task: PropTypes.shape({
		id: PropTypes.string.isRequired,
		title: PropTypes.string.isRequired,
		state: PropTypes.string.isRequired,
	}),
	onArchiveTask: PropTypes.func,
	onPinTask: PropTypes.func,
};
