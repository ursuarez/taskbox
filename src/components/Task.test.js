import React from 'react';
import { shallow } from 'enzyme';
import Task from './Task';
import { task, actions } from './Task.stories';

describe('Pruebas en <Task />', () => {
	test('debe de mostrar <Task /> correctamente ', () => {
		const wrapper = shallow(<Task task={task} {...actions} />);

		expect(wrapper).toMatchSnapshot();
	});
});
