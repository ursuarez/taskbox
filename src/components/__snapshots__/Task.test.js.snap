// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Task /> debe de mostrar <Task /> correctamente  1`] = `
<styled.div
  className="list-item TASK_INBOX"
>
  <label
    className="checkbox"
  >
    <input
      defaultChecked={false}
      disabled={true}
      name="checked"
      type="checkbox"
    />
    <span
      className="checkbox-custom"
      onClick={[Function]}
    />
  </label>
  <div
    className="title"
  >
    <input
      placeholder="Input title"
      readOnly={true}
      type="text"
      value="Test Task"
    />
  </div>
  <div
    className="actions"
    onClick={[Function]}
  >
    <a
      onClick={[Function]}
    >
      <span
        className="icon-star"
      />
    </a>
  </div>
</styled.div>
`;
